DIL ACM/read_int.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+// takes the token. Returns the base, or 0 if error
+int base_of(const char *);
+
+// takes ONE character and the base. returns -1 on error
+int read_int(char, const int);
+
+// takes the token and a variable to fill
+// with the integer. On error returns 0.
```

Comment:
What integer?

Reply:
To be more specific: In places like this, you don't want documentation that assumes that the reader knows what you're trying to do. For example, here "fill with the integer" assumes that the reader knows about what integer you're talking about. Also, the "token" is a bad term here, it means other things in compiler parlance and not what this function takes here.

A better doc could be e.g. (rename `token` to `input`): Extract the integer value from the integer literal found in `input` (assume that `input` starts with such a literal). Put the value in <second_param_name>.